### PR TITLE
Metrics on the progress of modularization

### PR DESCRIPTION
**Pull request by lapsedPacifist**
To measure the progress of modularisation we need metrics on the number of components created, the number and fraction of apps that have been componentized and the number and fraction of components that have a public API.

There maybe other metrics that need to be added (such as number of Oida violations and where they occur) in the future.

Closes: CORE-373

### CODE REVIEW

**Comment by lapsedPacifist** [comment]
Current JSON output for Tienda:
```json
{
    "components": {
        "number_of_components": 1,
        "number_of_components_with_public_API": 0,
        "percentage_of_components_with_public_API": 0.0,
        "components": [
            "products"
        ],
        "components_with_public_API": []
    },
    "apps": {
        "number_of_apps": 139,
        "number_of_apps_in_components": 4,
        "percentage_of_apps_in_components": 2.877697841726619,
        "componentized_apps": [
            "core",
            "units",
            "legalities",
            "local_info"
        ],
        "uncomponentized_apps": [
            "observed_prices",
            "recommendation",
            "model_properties",
            "backend_fulfillment_plan",
            "ads",
            "organizations",
            "price_areas",
            "supplier_edi",
            "user_addresses",
            "payments",
            "free_delivery",
            "data_pools",
            "cronjobs",
            "fabrica",
            "product_collections",
            "tasks",
            "recurring_orders",
            "rema_assortment_api",
            "editorial",
            "locale",
            "iterable",
            "core",
            "delivery_offering",
            "suppliers",
            "customer_subscriptions",
            "cart_monitor",
            "sms",
            "user_auth",
            "backend_reports",
            "lokalise_api",
            "geodata",
            "product_certifications",
            "stamp_cards",
            "internal_api",
            "mail",
            "user_segments",
            "referrals",
            "fraud",
            "experiments",
            "user_privacy",
            "leads",
            "delivery_safety_checks",
            "delivery_setup",
            "locking",
            "product_pricing",
            "product_dangerous_goods",
            "product_availability",
            "spoilages",
            "discounts",
            "meta",
            "backend_forecasting",
            "product_selection",
            "audit_logs",
            "third_party_shipments",
            "gunicorn",
            "supplier_transport",
            "delivery_capacity",
            "tango",
            "checkout",
            "giftcards",
            "supplier_deviations",
            "search",
            "fulfillment_centers",
            "backend_distribution",
            "driver_route_matching",
            "delivery_delays",
            "campaign_pages",
            "product_lists",
            "product_categorization",
            "coupons",
            "deliverables",
            "yard_control",
            "sessions",
            "samplings",
            "product_bom",
            "purchase_orders",
            "bulk_interventions",
            "support",
            "product_restrictions",
            "sitemaps",
            "user_preferences",
            "recipes",
            "urls",
            "settinglib",
            "deviations",
            "inventory",
            "user_communication",
            "delivery_routes",
            "federation",
            "users",
            "slot_picker",
            "slot_reservations",
            "route_planning",
            "relex",
            "cart",
            "supplier_integrations",
            "delivery",
            "db",
            "api",
            "supplier_contacts",
            "tracking",
            "pickuppoints",
            "user_crm",
            "finance",
            "request_logger",
            "third_party_marketing",
            "eyeshare",
            "delivery_tracker",
            "campaigns",
            "bottle_deposits",
            "user_permissions",
            "orders",
            "monitoring",
            "shopping_assistant",
            "trade_locations",
            "backend_fulfillment",
            "trade_items",
            "user_surveys",
            "front",
            "backend_invoices",
            "routes",
            "delivery_slots",
            "pat",
            "notifications",
            "delivery_fees",
            "climate_emissions",
            "app_components",
            "backend_products",
            "perks",
            "cloud_storage",
            "bravo",
            "pickups",
            "employees",
            "product_purchasing",
            "deliverable_products"
        ]
    }
}
```

**Comment by lapsedPacifist** [comment]
I first tried to get the apps from the `INSTALLED_APPS` configuration setting in tienda by importing the` settings.py`. But I ran into problems importing all the other modules that are imported by `settings.py`. These modules are not part of the virtual environment in which I run Oida... They might be when Oida is run as a linter from within Tienda, I guess, but that setup is difficult to test, and given little time I had, I decided, for now, to use the file system to find apps.

**Comment by lapsedPacifist** [comment]
> It seems like this is making a lot of assumptions about the structure of the codebase that it ideally shouldn't. For example services and selectors are part of what we've documented as defining for a component or app. Specifically in the [Concepts](https://github.com/kolonialno/oida#concepts) section of the readme it says:
> 
> > [...] A component is basically a collection of Django apps. [...]
> 
> I think that definition is really on point and if we follow that we can make the logic here much simpler:
> 
> 1. We need to know if a directory is a Django app. That would typically mean looking for files named `apps.py`, `admin.py`, `models.py`, `urls.py` or directories named `management`, `migrations`, `templates`, `static`. Maybe something like that?

Yes, I was struggling with how to detect Django apps - thanks for the pointers!

> 2. Detecting a component would then be just detecting if there's one or more django apps in the directory.

Good point.

File: oida/commands/linter.py [hunk 1]
```diff
@@ -11,9 +11,13 @@ def print_violation(
 
 
 def run_linter(*paths: Path, checks: list[str]) -> bool:
+    print(f"paths: {paths}")
+    # for path in paths:
+    #    statistics = StatisticsGenerator(path=path)
     has_violations = False
     checkers = get_checkers(checks)
     for module in find_modules(*paths):
+        print(f"module > {module.path}")
```
Comment (ljodal) [inline oida/commands/linter.py:20]: I'm assuming this is some leftovers that should be removed?

**Review by ljodal — APPROVED** [review]

File: oida/commands/linter.py (hunk 1 above)
Comment (lapsedPacifist) [reply oida/commands/linter.py]: Correct, I removed it just now.

**Pull request merged**